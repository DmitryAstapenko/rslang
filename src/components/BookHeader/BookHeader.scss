@import "../../scss/color-variables.scss";

.group-action {
  &-level-1 {
    background: $group-1-color !important;
  }
  &-level-2 {
    background: $group-2-color !important;
  }
  &-level-3 {
    background: $group-3-color !important;
  }
  &-level-4 {
    background: $group-4-color !important;
  }
  &-level-5 {
    background: $group-5-color !important;
  }
  &-level-6 {
    background: $group-6-color !important;
  }
}
