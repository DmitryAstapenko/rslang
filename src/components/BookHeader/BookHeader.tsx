import React from "react";
import Box from "@material-ui/core/Box";
import Typography from "@material-ui/core/Typography";

import "./BookHeader.scss";
import { Paper } from "@material-ui/core";

interface IBookHeaderProps {
  name: string;
  group: number;
}

const BookHeader: React.FunctionComponent<IBookHeaderProps> = ({
  name,
  group,
}: IBookHeaderProps) => {
  return (
    <Paper elevation={3} className={`group-action-level-${group + 1}`}>
      <Box marginTop={2} padding={2}>
        <Typography variant="h4" component="h2">
          {`${name} > Раздел ${group + 1}`}
        </Typography>
      </Box>
    </Paper>
  );
};

export default BookHeader;
