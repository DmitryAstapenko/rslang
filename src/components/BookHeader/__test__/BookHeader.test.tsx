import React from "react";
import BookHeader from "../BookHeader";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

interface IBookHeaderProps {
  name: string;
  group: number;
}
const setUp = (props:IBookHeaderProps) => shallow(<BookHeader {...props} />);

describe("should render BookHeader component", () => {
  let component:any;
  beforeEach(() => {
    const name: string = "";
    const group: number = 0;
    component = setUp({name, group});
  });
  it("should contain book-header", () => {
    const wrapper = component.find(".book-header");
    expect(wrapper.length).toBe(1);
  });
  it("should render BookHeader component", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
