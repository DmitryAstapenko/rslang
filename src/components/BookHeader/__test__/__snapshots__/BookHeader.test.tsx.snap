// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BookHeader component should render BookHeader component 1`] = `
<WithStyles(ForwardRef(Paper))
  className="group-action-level-1"
  elevation={3}
>
  <Styled(MuiBox)
    className="book-header"
    marginTop={2}
    padding={2}
  >
    <Styled(MuiBox)>
      <WithStyles(ForwardRef(Typography))
        component="h2"
        variant="h4"
      >
         &gt; Раздел 1
      </WithStyles(ForwardRef(Typography))>
    </Styled(MuiBox)>
    <Styled(MuiBox)>
      <BookSettings />
    </Styled(MuiBox)>
  </Styled(MuiBox)>
</WithStyles(ForwardRef(Paper))>
`;
